src/mm/memory_manager.rs

```diff
@@ -97,61 +149,121 @@ pub fn start_worker<VM: VMBinding>(
+///
+/// Arguments:
+/// * `mmtk`: The reference to an MMTk instance.
+/// * `tls`: The thread that enables the collection.
 pub fn enable_collection<VM: VMBinding>(mmtk: &'static MMTK<VM>, tls: OpaquePointer) {
     mmtk.scheduler.initialize(mmtk.options.threads, mmtk, tls);
     VM::VMCollection::spawn_worker_thread(tls, None); // spawn controller thread
     mmtk.plan.base().initialized.store(true, Ordering::SeqCst);
 }
 
+/// Process MMTk run-time options.
+///
+/// Arguments:
+/// * `mmtk`: The reference to an MMTk instance.
+/// * `name`: The name of the option.
+/// * `value`: The value of the option (as a string).
 pub fn process<VM: VMBinding>(mmtk: &'static MMTK<VM>, name: &str, value: &str) -> bool {
     unsafe { mmtk.options.process(name, value) }
 }
 
+/// Return used memory in bytes.
+///
+/// Arguments:
+/// * `mmtk`: The reference to an MMTk instance.
 pub fn used_bytes<VM: VMBinding>(mmtk: &MMTK<VM>) -> usize {
     mmtk.plan.get_pages_used() << LOG_BYTES_IN_PAGE
 }
 
+/// Return free memory in bytes.
+///
+/// Arguments:
+/// * `mmtk`: The reference to an MMTk instance.
 pub fn free_bytes<VM: VMBinding>(mmtk: &MMTK<VM>) -> usize {
     mmtk.plan.get_free_pages() << LOG_BYTES_IN_PAGE
 }
 
+/// Return the start adress of the heap. *Note that currently MMTk use
+/// a fixed address range as heap.*
 pub fn starting_heap_address() -> Address {
     HEAP_START
 }
 
+/// Return the end adress of the heap. *Note that currently MMTk use
+/// a fixed address range as heap.*
 pub fn last_heap_address() -> Address {
     HEAP_END
 }
 
+/// Return the total memory in bytes.
+///
+/// Arguments:
+/// * `mmtk`: The reference to an MMTk instance.
 pub fn total_bytes<VM: VMBinding>(mmtk: &MMTK<VM>) -> usize {
     mmtk.plan.get_total_pages() << LOG_BYTES_IN_PAGE
 }
 
+/// Perform a linear scan through a single contiguous region.
 #[cfg(feature = "sanity")]
+#[deprecated]
 pub fn scan_region() {
     crate::util::sanity::memory_scan::scan_region();
 }
 
+/// Trigger a garbage collection as requested by the user.
+///
+/// Arguments:
+/// * `mmtk`: The reference to an MMTk instance.
+/// * `tls`: The thread that triggers this colleciton request.
 pub fn handle_user_collection_request<VM: VMBinding>(mmtk: &MMTK<VM>, tls: OpaquePointer) {
     mmtk.plan.handle_user_collection_request(tls, false);
 }
 
+/// Is the object alive?
+///
+/// Arguments:
+/// * `object`: The object reference to query.
 pub fn is_live_object(object: ObjectReference) -> bool {
     object.is_live()
 }
 
+/// Is the object in the mapped memory?
+///
+/// Arguments:
+/// * `object`: The object reference to query.
 pub fn is_mapped_object(object: ObjectReference) -> bool {
     object.is_mapped()
 }
 
+/// Is the address in the mapped memory?
+///
+/// Arguments:
+/// * `address`: The address to query.
 pub fn is_mapped_address(address: Address) -> bool {
     address.is_mapped()
 }
 
+/// Check that if a garbage collection is in progress and if the given
```

Comment:
It's not clear what this function does.
Does it exit if `(a collection is in progress) && (object is movable)`?

If so, maybe it should be rewritten as:
Exit and log error messages if a garbage collection is in progress and the object is movable.

Reply:
I am not clear about this function either. It is ported from JikesRVM.